### DISCUSSION TITLE
Frequency stability of the high and low speed oscillators on the PYBD

### DISCUSSION THREAD

**Discussion by victorallume** [micropython/micropython#13523] [STM32 / Pyboard]
I'm trying to get some data on the frequency stability of the HSE and LSE oscillators on the PYBD, but can't find the exact part numbers used to work this out from the datasheets. Is there a BOM available anywhere, or can someone point me to the part numbers of the components used?

**Comment by dpgeorge** [comment]
The LSE is a SiT1533 (specifically SIT1533AC-H5-DCC-32.768E), link here: https://www.sitime.com/products/32-khz-oscillators/sit1533 (I'm pretty sure the part number is labelled on the schematics of the PYBD).  This is an accurate clock and should be preferred for taking timestamps.

The HSE is a simple 25MHz crystal with part number XRCGB25M000F3M00R0.

**Reply by victorallume** [reply]
Thanks for the info. 

We're generally using the LSE clock, but finding it sometimes either stops for a period of time, or skews (not certain if this is actually the case or if it's just stopping and re-starting). In that case we want to switch to using the HSE/ticks based clock as a fallback, but firstly need to determine if the LSE is faulty. But we first need to work out if the LSE is faulty - which we want to do by comparing the HSE and LSE values, and seeing if they differ significantly more than the maximum expected divergence. 

For the HSE, error is +/-30ppm max tolerance +/-40ppm max frequency shift by temperature, and +/-5ppm max./year aging. Assuming worst case for a device 10 years old, this gives +/-120 ppm

For the LSE, the datasheet gives +/- 20ppm tolerance, plus a 75ppm max frequency shift by temperature. So I'll round that to 100ppm worst case.

First up, I'm surprised that the tolerances are quite similar - I would have expected the LSE to be substantially more precise than the HSE.

Adding the two errors together, let's round it to 200ppm. Which is 0.02%. Much less than I would expect. 

Experimentally comparing the results of RTC-based vs CPU ticks based time, I'm getting an error of 4% on one pyboard, and 2.6% on another. Significantly more than my calculated maximum. 

Any thoughts? Are my calculations out? Have I read the datasheets wrong?

**Reply by victorallume** [reply]
So I've worked out that the 4% error is because the board was using the LSI instead of the LSE in my test case. Still trying to work out why the LSE isn't reliably working (it works some of the time, and I'm getting a good signal from the oscillator when I probe it on the breakout board).
